libs/ml/include/ml/model/model.hpp

```diff
@@ -17,19 +17,16 @@
 //
 //------------------------------------------------------------------------------
 
-#include "ml/core/graph.hpp"
 #include "ml/dataloaders/tensor_dataloader.hpp"
```

Comment:
you need to move most of these remaining includes into the source file

Reply:
I removed as much as possible. types are used in function signatures, graph_saver is used in serialization

Reply:
what about forward declaration for types used in function signatures, and moving serialisation code to serialisers/ml_types to avoid including graph_saver?

Reply:
Done